tests/backend/test_kv_store.py

```diff
@@ -0,0 +1,177 @@
+import pytest
+from unittest.mock import MagicMock, call
+
+from prefect.backend.kv_store import set_key_value, get_key_value, delete_key, list_keys
+from prefect.utilities.exceptions import ClientError
+from prefect.utilities.graphql import GraphQLResult
+
+
+class TestSetKeyValue:
+    def test_set_key_value_raises_on_server_backend(self, server_api):
+        with pytest.raises(ClientError):
+            set_key_value(key="foo", value="bar")
+
+    def test_set_key_value_calls_client_mutation_correctly(
+        self, monkeypatch, cloud_api
+    ):
+        gql_return = MagicMock(
+            return_value=MagicMock(
+                data=GraphQLResult(
+                    set_key_value=GraphQLResult({"id": "123"}),
+                )
+            )
+        )
+        client = MagicMock()
+        client.return_value.graphql = gql_return
```

Comment:
You don't have to do this; it's frustrating that it's done so often in our tests

See an example of a nicer way at https://github.com/PrefectHQ/prefect/blob/3d8c4c943627c2c41ce8ffc2d90c720c330c7fc3/tests/cli/test_run.py#L626

I would write this as something like...
```python
        Client = MagicMock()
        Client().graphql.return_value = GraphQLResult(
            data={"set_key_value": {"id": "123"}}
        ) 
        monkeypatch.setattr("prefect.backend.kv_store.Client", Client)

        key_value_id = set_key_value(key="foo", value="bar")
        Client().graphql.assert_called_with(
            query={
                "mutation($input: set_key_value_input!)": {
                    "set_key_value(input: $input)": {"id"}
                }
            },
            variables={"input": {"key": "foo", "value": "bar"}},
        )
        assert key_value_id == "123"
```

`GraphQLResult` is a `Box` subclass so it'll roll all of the subdictionaries into the `Box` as well 🤷

Reply:
good point, I copied one test case and re-used the logic without really examining it here, updated